packages/client/hmi-client/src/petrinet/petrinet-renderer.ts

```diff
@@ -275,11 +275,15 @@ export class PetrinetRenderer extends BasicRenderer<NodeData, EdgeData> {
 		this.removeAllEvents('node-drag-end');
 
 		// (Re)create dragging listeners
-		this.on('node-drag-start', (_eventName, _event, selection: D3SelectionINode<NodeData>) => {
+		this.on('node-drag-start', (_eventName, event, selection: D3SelectionINode<NodeData>) => {
 			if (!this.isDragEnabled) return;
 			sourceData = selection.datum();
 			start.x = sourceData.x;
 			start.y = sourceData.y;
+
+			const target = event.sourceEvent.target;
+			start.x += +d3.select(target).attr('cx');
+			start.y += +d3.select(target).attr('cy');
```

Comment:
Why not? Just out of curiosity?
```suggestion
			const target = d3.select(event.sourceEvent.target);
			start.x += +target.attr('cx');
			start.y += +target.attr('cy');
```

Reply:
No particular reason other than I was trying to use target.getBBox() in an earlier attempt.

Reply:
changed